### DISCUSSION TITLE
Deserialize something into `HashMap<Cow<'a, str>, MyStruct<;'a>>`

### DISCUSSION THREAD

**Discussion by mainrs** [jonasbb/serde_with#597] [Q&A] [answered]
I am not entirely sure if this makes even that much sends due to `HashMap` key boundaries. But I'd like to have my `HashMap` keys either borrowed or, if not possible, be owned.

But the following code does not work:

```rust
#[serde_as(as = "HashMap<BorrowCow, MyStruct<'a>>")]
pub entries: HashMap<Cow<'a, str>, MyStruct<'a>>,
```

Is this possible?

**Comment by jonasbb** [comment, marked as answer]
You are missing some key details in your question, so it is hard to answer. If you provide a small compilable example, that makes is much easier to see what is going on and, very importantly, shows the error messages you get.

I think you might have the problem from https://github.com/jonasbb/serde_with/issues/202.
In that case, the correct attribute is `#[serde_as(as = "HashMap<BorrowCow, _>")]`.
